android/gltfio-android/src/main/java/com/google/android/filament/gltfio/AssetLoader.java

```diff
@@ -152,6 +152,32 @@ public FilamentAsset createInstancedAsset(@NonNull Buffer buffer,
         return new FilamentAsset(mEngine, nativeAsset);
     }
 
+    /**
+     * Adds a new instance to an instanced asset.
+     *
+     * Use this with caution. It is more efficient to pre-allocate a max number of instances, and
+     * gradually add them to the scene as needed. Instances can also be "recycled" by removing and
+     * re-adding them to the scene.
+     *
+     * NOTE: destroyInstance() does not exist because gltfio favors flat arrays for storage of
```

Comment:
Would it be that slow? Those arrays won't be big.

Reply:
Not super slow, but in some cases, the number of entities in an instance can be large.  Also the texture dependency graph has to be re-checked in case a large texture are still being loaded during instance creation.